src/Transit/Internal/FileTransfer.hs

```diff
@@ -96,6 +101,14 @@ send session appid password printHelpFn tfd = do
             return ()
     )
 
+sendPipeline :: C.MonadResource m =>
+                FilePath
+             -> TCPEndpoint
+             -> SecretBox.Key
+             -> C.ConduitM a c m (Hash.Digest SHA256, ())
+sendPipeline fp (TCPEndpoint s) key =
+  C.sourceFile fp .| C.takeC 4096 .| sha256PassThroughC `C.fuseBoth` (encryptC key .| CN.sinkSocket s)
```

Comment:
So far as I can understand the docs, `takeC 4096` will await the first 4096 bytes of the source file and make them available for the next component of the pipeline (`sha256PassthroughC`) but it won't make any subsequent bytes available.

That doesn't seem like what we want here.  Did I misunderstand how this works?  This interaction seems to support this interpretation:

```
C DC> runConduit (yieldMany [1..20] .| takeC 5 .| sinkList)
[1,2,3,4,5]
```

Reply:
My intention to use a `takeC` was to chunk the stream into blocks of 4096 bytes. I think it worked in my case (I tested with large files as well) because the upstream element was a filesource, the handle was open and it was waiting for the downstream to ask for more data in a loop to yield output.

But your comment made me relook the entire popeline. I read through the `sourceFile` code. It looks like there is this constant called `defaultChunkSize` in the `bytestring` package (`Data.ByteString.Lazy.Internal`) that conveniently defines it to 32k and the `sourceFile` is using it to yield a chunk.

https://hackage.haskell.org/package/conduit-1.3.0.3/docs/src/Data.Conduit.Combinators.html#sourceFile

So, perhaps we don't need the `takeC` combinator there at all. I tried it and it seem to work just fine.